web/shared/src/typeahead.ts

```diff
@@ -193,14 +194,25 @@ export const parse_unicode_emoji_code = (code: string): string =>
         .map((hex) => String.fromCodePoint(Number.parseInt(hex, 16)))
         .join("");
 
-export function get_emoji_matcher(query: string): (emoji: Emoji) => boolean {
+// TODO: can this file not import from web/src?
+export type EmojiPillData = {
+    reaction_type: string;
+    emoji_name: string;
+    emoji_url?: string;
+    emoji_code?: string;
+    is_realm_emoji: boolean;
+    type: "emoji";
+};
+
+export function get_emoji_matcher(query: string): (emoji: EmojiPillData) => boolean {
     // replace spaces with underscores for emoji matching
     query = query.replace(/ /g, "_");
     query = clean_query_lowercase(query);
 
     return function (emoji) {
         const matches_emoji_literal =
             emoji.reaction_type === "unicode_emoji" &&
+            emoji.emoji_code !== undefined &&
```

Comment:
`emoji_code` should always be set for `reaction_type: "unicode"` emojis, but `EmojiPillData` doesn't capture that. I should probably dive a bit more into this and consolidate the types. The type `Emoji` in this file doesn't have a `emoji_url` and it does in `composebox_typeahead` so that's something to look into next.